ui/StatusQ/src/keychain.cpp

```diff
@@ -0,0 +1,81 @@
+#include "StatusQ/keychain.h"
+
+#include <QDebug>
+
+Keychain::Keychain(QObject *parent) : QObject(parent)
+{}
+
+#ifndef Q_OS_MACOS
+Keychain::~Keychain() = default;
+#endif
+
+QString Keychain::service() const
+{
+    return m_service;
+}
+
+void Keychain::setService(const QString &service)
+{
+    if (m_service == service)
+        return;
+
+    m_service = service;
+    emit serviceChanged();
+}
+
+QString Keychain::reason() const
+{
+    return m_reason;
+}
+
+void Keychain::setReason(const QString &reason)
+{
+    if (m_reason == reason)
+        return;
+
+    m_reason = reason;
+    emit reasonChanged();
+}
+
+bool Keychain::loading() const
+{
+    return m_loading;
+}
+
+void Keychain::setLoading(bool loading)
+{
+    if (m_loading == loading)
+        return;
+
+    m_loading = loading;
+    emit loadingChanged();
+}
+
+#ifndef Q_OS_MACOS
+void Keychain::requestSaveCredential(const QString& account, const QString& password)
+{
+    Q_UNUSED(account);
+    Q_UNUSED(password);
+    qWarning() << "Keychain::requestSaveCredential is intended to be called only on MacOS.";
+    emit saveCredentialRequestCompleted(Keychain::StatusNotSupported);
```

Comment:
I think it's not necessary to emit anything in that case. We can assume it's just not called at all. When it's emitted, for consistency all handlers should check and filter out cases with `StatusNotSupported`.

Reply:
@micieslak Thing is that otherwise there's no trace of the error anywhere except the logs 🤔 
If we make a mistake in the future and use `Keychain` on non-supported platforms, would be nice to actually get the error.
Another option is to return `bool` from this method, but this seemed inconsistent to me.

Reply:
@micieslak why do we need to handle this specific error in UI?
1. We know that it will never occur, because we won't use `Keychain` on non-macs.
2. For the worst case when we fail with (1), we can handle this specific error same as `GenericError`. Otherwise nobody will ever see this error.

When thinking of the `Keychain` component, it's bad to _just log_ the issue and never notify the user through the API. The API should be consistent across platforms. And the API says "call `async...` and expect `...Completed`.

Reply:
> We know that it will never occur, because we won't use Keychain on non-macs.
For that reason there is no need to try recover from that situation.

But I understand your point, with emitting the signal there is still a chance to handle that situation relatively "nicely", whatever it means for sth that should never happen.